course/src/main/java/org/openedx/course/presentation/ui/CourseUI.kt

```diff
@@ -573,81 +580,184 @@ fun VideoSubtitles(
+                    onClick = { onDownloadClick() }) {
+                    Icon(
+                        painter = downloadIconPainter,
+                        contentDescription = downloadIconDescription,
+                        tint = downloadIconTint
+                    )
+                }
+            } else if (downloadedState != null) {
+                Box(contentAlignment = Alignment.Center) {
+                    if (downloadedState == DownloadedState.DOWNLOADING) {
+                        CircularProgressIndicator(
+                            modifier = Modifier.size(28.dp),
+                            backgroundColor = Color.LightGray,
+                            strokeWidth = 2.dp,
+                            color = MaterialTheme.appColors.primary
+                        )
+                    } else if (downloadedState == DownloadedState.WAITING) {
+                        Icon(
+                            painter = painterResource(id = R.drawable.course_download_waiting),
+                            contentDescription = stringResource(id = R.string.course_accessibility_stop_downloading_course_section),
+                            tint = MaterialTheme.appColors.error
+                        )
+                    }
+                    IconButton(
+                        modifier = iconModifier.padding(2.dp),
+                        onClick = { onDownloadClick() }) {
+                        Icon(
+                            imageVector = Icons.Filled.Close,
+                            contentDescription = stringResource(id = R.string.course_accessibility_stop_downloading_course_section),
+                            tint = MaterialTheme.appColors.error
+                        )
+                    }
+                }
+            }
         }
     }
 }
 
 @Composable
 fun CourseSubSectionItem(
-    modifier: Modifier,
+    modifier: Modifier = Modifier,
     block: Block,
-    downloadedState: DownloadedState?,
-    downloadsCount: Int,
-    onClick: (Block) -> Unit,
-    onDownloadClick: (Block) -> Unit
+    onClick: (Block) -> Unit
 ) {
     val icon =
-        if (block.isCompleted()) painterResource(R.drawable.course_ic_task_alt) else painterResource(
-            coreR.drawable.ic_core_chapter_icon
-        )
+        if (block.isCompleted()) painterResource(R.drawable.course_ic_task_alt) else painterResource(coreR.drawable.ic_core_chapter_icon)
     val iconColor =
-        if (block.isCompleted()) MaterialTheme.appColors.primary else MaterialTheme.appColors.onSurface
+        if (block.isCompleted()) MaterialTheme.appColors.successGreen else MaterialTheme.appColors.onSurface
 
-    val iconModifier = Modifier.size(24.dp)
-
-    Column(Modifier
-        .clickable { onClick(block) }
-        .background(if (block.isCompleted()) MaterialTheme.appColors.surface else Color.Transparent)
+    val due = block.due?.let { TimeUtils.getAssignmentFormattedDate(LocalContext.current, it) }
```

Comment:
If I'm not mistaken, it's better to use remember when you transform dates in compose because it creates a calendar instance on each recomposition.